src/app/models/admin_verification.server.model.ts

```diff
@@ -0,0 +1,98 @@
+import { Mongoose, Schema } from 'mongoose'
+
+import { IUserSchema } from 'src/types'
+
+import {
+  IAdminVerificationModel,
+  IAdminVerificationSchema,
+  UpsertOtpParams,
+} from '../../types/admin_verification'
+
+import { USER_SCHEMA_ID } from './user.server.model'
+
+export const ADMIN_VERIFICATION_SCHEMA_ID = 'AdminVerification'
+
+const AdminVerificationSchema = new Schema<IAdminVerificationSchema>(
+  {
+    admin: {
+      type: Schema.Types.ObjectId,
+      ref: USER_SCHEMA_ID,
+      required: 'AdminVerificationSchema must have an Admin',
+    },
+    hashedContact: {
+      type: String,
+      required: true,
+    },
+    hashedOtp: {
+      type: String,
+      required: true,
+    },
+    expireAt: {
+      type: Date,
+      required: true,
+    },
+    numOtpAttempts: {
+      type: Number,
+      default: 0,
+    },
+    numOtpSent: {
+      type: Number,
+      default: 0,
+    },
+  },
+  {
+    timestamps: true,
+  },
+)
+AdminVerificationSchema.index({ expireAt: 1 }, { expireAfterSeconds: 0 })
+
+// Statics
+/**
+ * Upserts given OTP into AdminVerification collection.
+ */
+AdminVerificationSchema.statics.upsertOtp = async function (
+  this: IAdminVerificationModel,
+  upsertParams: UpsertOtpParams,
+) {
+  return this.findOneAndUpdate(
+    { admin: upsertParams.admin },
+    {
+      $set: { ...upsertParams, numOtpAttempts: 0 },
+      $inc: { numOtpSent: 1 },
+    },
+    { upsert: true, new: true, setDefaultsOnInsert: true, runValidators: true },
```

Comment:
should we enable `runValidators` on `mongoose` globally? https://mongoosejs.com/docs/api/mongoose.html#mongoose_Mongoose-set

Reply:
@liangyuanruo